dotCMS/src/main/java/com/dotcms/rendering/velocity/services/VTLLoader.java

```diff
@@ -0,0 +1,144 @@
+    private static String VELOCITY_ROOT;
+    private static VTLLoader vtlServices;
+
+    private VTLLoader() {
+        init();
+    }
+
+    public static VTLLoader instance() {
+        if (vtlServices == null) {
+            synchronized (VTLLoader.class) {
+                if (vtlServices == null) {
+                    vtlServices = new VTLLoader();
+                }
+            }
+        }
+        return vtlServices;
+    }
+
+    private void init() {
+        String velocityRootPath = Config.getStringProperty("VELOCITY_ROOT", "/WEB-INF/velocity");
+        if (velocityRootPath.startsWith("/WEB-INF")) {
+            String startPath = velocityRootPath.substring(0, 8);
+            String endPath = velocityRootPath.substring(9, velocityRootPath.length());
+            velocityRootPath = FileUtil.getRealPath(startPath) + File.separator + endPath;
+        } 
+
+        VELOCITY_ROOT = velocityRootPath + File.separator;
+
+        File f = new File(VELOCITY_ROOT);
+        try {
+            if(f.exists()){
+                   velocityCanoncalPath = f.getCanonicalPath();
+            }
+        } catch (IOException e) {
+            Logger.fatal(this,e.getMessage(),e);
+        }
+
+        try {
+            if(UtilMethods.isSet(Config.getStringProperty("ASSET_REAL_PATH"))){
+                f = new File(Config.getStringProperty("ASSET_REAL_PATH"));
+                if(f.exists()){
+                        assetRealCanoncalPath = f.getCanonicalPath();
+                }
+            }
+        } catch (IOException e) {
+            Logger.fatal(this,e.getMessage(),e);
+        }
+
+        try {
+            if(UtilMethods.isSet(Config.getStringProperty("ASSET_PATH"))){
+                f = new File(FileUtil.getRealPath(Config.getStringProperty("ASSET_PATH")));
+                if(f.exists()){
+                    assetCanoncalPath = f.getCanonicalPath();
+                }
+            }
+        } catch (IOException e) {
+            Logger.fatal(this,e.getMessage(),e);
+        }
+
+
+    }
+
+    InputStream streamFile(String filePath) throws IOException {
+        boolean serveFile = false;
+        Logger.debug(this, "Not a CMS Velocity File : " + filePath);
+
+        java.io.File f = null;
+        String lookingFor = "";
+        if (filePath.startsWith("dynamic")) {
+            lookingFor = ConfigUtils.getDynamicContentPath() + File.separator + "velocity" + File.separator + filePath;
+
+        } else {
+            lookingFor = VELOCITY_ROOT + filePath;
+        }
+        f = new java.io.File(lookingFor);
+        if (!f.exists()) {
+            f = new java.io.File(filePath);
+        }
+        if (!f.exists()) {
+            throw new ResourceNotFoundException("cannot find resource");
+        }
+        String canon = f.getCanonicalPath();
+        File dynamicContent = new File(ConfigUtils.getDynamicContentPath());
+
+        if (assetRealCanoncalPath != null && canon.startsWith(assetRealCanoncalPath)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this conditional structure or edit its code blocks so that they're not all the same. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3923)